File: 4 - Estrutura de Repetição/Exercicio.10.py
#10. Faça um algoritmo para um intervalo de números informados pelo usuário
# e calcular, para cada número, a raiz quadrada e sua metade.


limite_inferior = int(input("Digite o limite inferior: "))
limite_superior = int(input("Digite o limite superior: "))

for numero in range(limite_inferior, limite_superior + 1):
    raiz_quadrada = numero ** 0.5
    metade = numero / 2
    print(f"Para {numero}: Raiz quadrada = {raiz_quadrada:.2f}, Metade = {metade}")

File: Livro - Módulo Java/4 - Estrutura de Repetição/Exercicio10.java
import java.util.Scanner;

public class Exercicio10 {
    public static void main(String[] args) {
        // Faça um algoritmo para um intervalo de números informados pelo usuário e calcular, para cada número, a raiz quadrada e sua metade

        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite o limite inferior do intervalo: ");
        double limiteInferior = scanner.nextDouble();

        System.out.print("Digite o limite superior do intervalo: ");
        double limiteSuperior = scanner.nextDouble();

        calcularRaizQuadradaEMetadeNoIntervalo(limiteInferior, limiteSuperior);
    }

    public static void calcularRaizQuadradaEMetadeNoIntervalo(double limiteInferior, double limiteSuperior) {
        System.out.println("Resultados:");

        for (double i = limiteInferior; i <= limiteSuperior; i++) {
            double raizQuadrada = calcularRaizQuadrada(i);
            double metade = calcularMetade(i);

            System.out.println("Número: " + i);
            System.out.println("Raiz Quadrada: " + raizQuadrada);
            System.out.println("Metade: " + metade + "\n");
        }
    }

    public static double calcularRaizQuadrada(double numero) {
        return Math.sqrt(numero);
    }

    public static double calcularMetade(double numero) {
        return numero / 2;
    }
}